test/with-typescript/index.ts

```diff
@@ -1933,3 +1934,20 @@ class NextResponse {}
 NextJS.getAppDirRequestHandler(NextResponse);
 
 customVerifySession({ checkDatabase: true })(new PreParsedRequest({} as any), new CollectingResponse());
+
```

Comment:
you can use the actual NextJS response and request object in this playground and pass them into withSession / getSSRSession to check that don't have any type issues. Depending on nextjs in this playground is fine